datafusion/physical-plan/src/sorts/sort.rs

```diff
@@ -1158,7 +1153,10 @@ impl ExecutionPlan for SortExec {
                     context.session_config().batch_size(),
                     context.runtime_env(),
                     &self.metrics_set,
-                    self.filter.clone(),
+                    self.filter
+                        .as_ref()
+                        .expect("Filter should be set when fetch is Some")
+                        .clone(),
```

Comment:
I refactored so that the `TopK` struct always expects this parameter which better reflects the reality of execution. But since it's strangely tied to the `fetch` param I am doing an `expect` assertion here. It should never fail at runtime.